lib/bbbevents/recording.rb

```diff
@@ -86,16 +90,36 @@ def create_csv(filepath)
       end
     end
 
+    def attendees_to_array
+      att_array = Array.new
+      @attendees.values.map { |att| 
+        att_array.push(att.to_hash)
+      }
+      att_array
+    end
+
+    def polls_to_array
+      polls_array = Array.new
+      @polls.values.map { |poll| 
+        polls_array.push(poll.to_hash)
+      }
+      polls_array
+    end
+
     def to_json
+      # Convert into an array so we can generate proper json
+      att_array = attendees_to_array
+      polls_array = polls_to_array
+
       {
         metadata: @metadata,
         meeting_id: @meeting_id,
         duration: @duration,
         start: @start,
         finish: @finish,
-        attendees: attendees.map(&:to_json),
+        attendees: att_array, 
```

Comment:
Does this change the structure of the final json output? Also, there's no reason to have the attendees_to_array, etc. functions. For example, attendees_to_array is equivalent to changing this line to
```ruby
   attendees: attendees.map(&:to_h)
```
(assuming you also rename the `to_hash` function to `to_h` like my earlier recommendation)

Reply:
Changed to `attendees.map(&:to_h)`. Same with polling.